sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -467,12 +467,8 @@ object JdbcUtils extends Logging with SQLConfHelper {
 
     case StringType if metadata.contains("rowid") =>
       (rs: ResultSet, row: InternalRow, pos: Int) =>
-        val rawRowId = rs.getRowId(pos + 1)
-        if (rawRowId == null) {
-          row.update(pos, null)
-        } else {
-          row.update(pos, UTF8String.fromString(rawRowId.toString))
-        }
+        val id = nullSafeConvert[RowId](rs.getRowId(pos + 1), r => UTF8String.fromBytes(r.getBytes))
```

Comment:
Does this assume that `rowid` column is always UTF-8?

Reply:
Maybe, could you add a reference Java doc link to ensure that assumption?

Reply:
The assumption might be practical as JDBC clients use the platform's default charset. However, I guess we cannot make such an assumption. I might close this first to see if we can apply the charset suitably for the client-server encode conversion.

Reply:
When I say 'practical', it means RowIds are alphanumerics

Reply:
Got it. Thank you for the further investigation to make this sure.